rollrus.go

```diff
@@ -2,60 +2,120 @@ package rollrus
 
 import (
 	"fmt"
+	"io"
 	"os"
+	"runtime"
+	"sync"
 	"time"
 
+	"github.com/benjamindow/rollrus/buffer"
+	"github.com/benjamindow/rollrus/buffer/channel"
 	log "github.com/sirupsen/logrus"
 	"github.com/stvp/roll"
 )
 
+type noopCloser struct{}
+
+func (c noopCloser) Close() error {
+	return nil
+}
+
+type RollrusConfig struct {
+	Buffer     buffer.Buffer
+	NumWorkers int
+	LogLevels  []log.Level
+}
+
 var defaultTriggerLevels = []log.Level{
 	log.ErrorLevel,
 	log.FatalLevel,
 	log.PanicLevel,
 }
 
+var defaultNumWorkers = 8 * runtime.NumCPU()
+var defaultBufferSize = 2 * defaultNumWorkers
+
 // Hook wrapper for the rollbar Client
 // May be used as a rollbar client itself
 type Hook struct {
 	roll.Client
 	triggers []log.Level
+	entries  buffer.Buffer
+	closed   chan struct{}
+	once     *sync.Once
+	wg       *sync.WaitGroup
+	pool     chan chan job
 }
 
 // Setup a new hook with default reporting levels, useful for adding to
 // your own logger instance.
 func NewHook(token string, env string) *Hook {
-	return NewHookForLevels(token, env, defaultTriggerLevels)
+	return NewHookForLevels(token, env, RollrusConfig{})
 }
 
 // Setup a new hook with specified reporting levels, useful for adding to
 // your own logger instance.
-func NewHookForLevels(token string, env string, levels []log.Level) *Hook {
-	return &Hook{
+func NewHookForLevels(token string, env string, config RollrusConfig) *Hook {
+	if len(config.LogLevels) == 0 {
```

Comment:
Might consider moving all of this into the config struct as an unexported method